src/Rating.js

```diff
@@ -1,8 +1,190 @@
 'use strict';
 
-import RatingBase from './Rating.soy';
+import core from 'metal';
+import Component from 'metal-component';
+import Soy from 'metal-soy';
+import templates from './Rating.soy';
 
-class Rating extends RatingBase {
+class Rating extends Component {
+    /**
+     * @inheritDoc
+     */
+    constructor(opt_config) {
+        super(opt_config);
+
+        this.ratingClicked = -1;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    attached() {
+        this.delegate('mouseover', '.rating-item', this.handleMouseOverEvent.bind(this));
+        this.delegate('click', '.rating-item', this.handleClickEvent.bind(this));
+        this.on('mouseleave', this.handleMouseLeaveEvent.bind(this));
+    }
+
+    /**
+     * Handles click event
+     * @param {Event} event
+     * @protected
+     */
+    handleClickEvent(event) {
+        if (!this.disabled) {
+            let index = parseInt(event.delegateTarget.dataset.index, 10);
```

Comment:
It seems like `data-set` [doesn't work on IE9](http://caniuse.com/#feat=dataset), so you should replace its calls with `getAttribute('data-index')` instead.